src/Engine/Utils.java

```diff
@@ -671,6 +675,15 @@ public static boolean enqueueMoveEvent(MapMaster gameMap, Unit unit, GamePath mo
       movePath.snipCollision(gameMap, unit);
       originalPathOK = false;
     }
+    if( unit.model.needsFuel ) // Check fuel.
+    {
+      int fuelBurn = movePath.getFuelCost(unit, gameMap);
+      if( fuelBurn > unit.fuel )
+      {
+        movePath.snipForFuel(gameMap, unit);
+        originalPathOK = false;
+      }
```

Comment:
Are we ever calling `enqueueMoveEvent()` with paths that exceed the unit's range? I thought we just used this to handle the uncertainty of ambushes on FoW; fuel (and anything else we could know a priori) should be (and was, I thought) checked before this point.

Reply:
That validation was in `MoveEvent`; I just moved most of it here so I can do terrible misdeeds with Kaiju movement (the preconditions will be satisfied by the time the `MoveEvent` *executes*, but not at the time it's created/queued).

There's also a test that fails if this validation isn't in the loop somewhere.

Reply:
The subject of whether that validation ought to happen at this point is harder to answer - the player probably can't violate anything here, but I think I'd prefer to not trust AIs with unbridled movement speed.

Reply:
> the preconditions will be satisfied by the time the MoveEvent executes, but not at the time it's created/queued

Might have to unpack this one for me. Terrible misdeeds, eh? Also, if the preconditions aren't satisfied at creation time, won't this snip the route so it doesn't do the full move anyway?

> The subject of whether that validation ought to happen at this point is harder to answer
I don't think I object to moving the validation here from `MoveEvent` though. Now that it's the main entrypoint for creating moves, I think that makes sense.

Reply:
> Might have to unpack this one for me. Terrible misdeeds, eh?

Since I know I will be stomping away any unit on a tile I move through, I can build a `MoveEvent` onto or even from a tile that already has a different unit on it.
> Also, if the preconditions aren't satisfied at creation time, won't this snip the route so it doesn't do the full move anyway?

It would, if I was using this helper function. Which is why I moved the validation here - so I can circumvent it when I have other ways of ensuring the gamestate stays legal.

Reply:
Are you creating/enqueuing `MoveEvent`s somewhere besides this helper function? I'm not seeing where, if so. I'm not seeing it. Or is that just later?

Reply:
That's in the Kaiju movement function, yes.